2D_model_LES_sin.jl

```diff
@@ -0,0 +1,225 @@
+#using Pkg
+using Oceananigans
+using Printf
+using Statistics
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.OutputReaders: FieldTimeSeries
+using Oceananigans.Grids: xnode, ynode, znode
+using SeawaterPolynomials
+using SeawaterPolynomials:TEOS10
+import SeawaterPolynomials.TEOS10: s, ΔS, Sₐᵤ
+s(Sᴬ::Number) = Sᴬ + ΔS >= 0 ? √((Sᴬ + ΔS) / Sₐᵤ) : NaN
```

Comment:
I think this is an interesting idea. Just need to use non-short-circuiting logic and be mindful of number type and it could potentially go in `SeawaterPolynomials`

Reply:
I was writing this in order to address the issue of 
```julia
ERROR: DomainError with -2.0:
sqrt was called with a negative real argument but will only return a complex result if called with a complex argument. Try sqrt(Complex(x)).
```
when I was training my model, in order to continue the training with `NaN`. 
Wouldn't doing non-short-circuiting logic mean that we are forced into a `DomainError` if `Sᴬ + ΔS` becomes negative?

Reply:
Hmm, we could clip to avoid it, for example:

```julia
function s(Sᴬ)
    not_positive = Sᴬ + ΔS >= 0
    d = max(zero(Sᴬ), Sᴬ + ΔS)
    return ifelse(not_positive, NaN, √(d / Sₐᵤ))
end
```

but after reflecting on this a bit more, I think it's better to receive a `DomainError` from this point, than to get a NaN and not know why.

Reply:
I agree. I believe I started doing this to throw `NaN`s when using EKI, which was necessary to continue training, but apart from training purposes there shouldn't be a need to do this

Reply:
Ahh no but I see your point now.

The thing is, we really do want to support training / automatic calibration and it shouldn't require hacks like this, I feel this really impacts reproducibility and understandability (technically this is type piracy...)

Maybe clipping+NaN should be an option of the equation of state then.

Reply:
Another option would be to wrap `run!(simulation)` inside a `try/catch` so that, if a simulation errors, we can replace the output with NaN / mark the simulation as failed for the purpose of estimating parameters. This would work too right?

Reply:
I suspect what one could do is throw `NaN` and a warning message that this is due to the model having salinity values that are beyond the regime where TEOS-10 is correct.

Reply:
We can't warn from inside a kernel though.

Reply:
ah, so try-catch it is then